docs/public/s2i-to-devfile.adoc

```diff
@@ -0,0 +1,92 @@
+== Migrating from S2I components to Devfile components
+
+Devfiles are new way of deploying component with odo that offers more convenience and flexibility. Read more about devfiles https://devfile.github.io/[here]
+
+Before 2.0 release odo deploys component with a different mechanism called as s2i based components. The newer way of deploying components is with devfiles.
+
+Exisiting users using s2i based components could migrate to devfile component by following the below steps:
+
+. Change the current directory to the component directory you want to migrate.
++
+[source,sh]
+----
+ $ cd <directory-name>
+----
+
+. Check the component status 
++
+[source,sh]
+----
+ $ odo list
+
+Openshift Components: 
+APP     NAME      PROJECT        TYPE       SOURCETYPE     STATE
+app     myapp     default        nodejs     local          Pushed
+
+----
+
+. Convert s2i component to devfile component, it would convert the s2i based configuration files to devfile configuration files.
++
+[source,sh]
+----
+ $ odo utils convert-to-devfile
+ 
+ devfile.yaml is available in the current directory.
+
+----
+
+. Push the devfile component to the cluster
++
+[source,sh]
+----
+ $ odo push 
+ 
+ Validation
+  ✓  Validating the devfile [40136ns]
+
+ Creating Kubernetes resources for component myapp
+  ✓  Waiting for component to start [12s]
+
+ Applying URL changes
+  ✓  URL myapp-8080: http://myapp-8080-myapp-default.apps-crc.testing/ created
+
+ Syncing to component myapp
+  ✓  Checking files for pushing [499419ns]
+  ✓  Syncing files to the component [155ms]
+
+ Executing devfile commands for component myapp
+  ✓  Executing s2i-assemble command "/opt/odo/bin/s2i-setup && /opt/odo/bin/assemble-and-restart" [47s]
+  ✓  Executing s2i-run command "/opt/odo/bin/run" [1s]
+
+ Pushing devfile component myapp
+  ✓  Changes successfully pushed to component
+
+----
+
+. Check the component status, at this point two components are running s2i based and devfile based.
++
+[source,sh]
+----
+ $ odo push 
+ 
+ Devfile Components: 
+ APP     NAME      PROJECT        TYPE      STATE
+ app     myapp     rejreromll     myapp     Pushed
```

Comment:
what's `rejreromll` ? Use a different project name such as `myproject`

you should use something neutral